src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -186,31 +189,46 @@ public void setupSchema(List<ColumnMetadata> columns) {
      */
     if (clientBufferParameters.getIsIcebergMode()) {
       for (ColumnDescriptor columnDescriptor : schema.getColumns()) {
-        String columnPath = concatDotPath(columnDescriptor.getPath());
+        String[] path = columnDescriptor.getPath();
+        String columnDotPath = concatDotPath(path);
         PrimitiveType primitiveType = columnDescriptor.getPrimitiveType();
+        boolean isInRepeatedGroup =
+            path.length > 1
+                && schema
+                    .getType(Arrays.copyOf(path, path.length - 1))
+                    .isRepetition(Type.Repetition.REPEATED);
 
         /* set fieldId to 0 for non-structured columns */
-        int fieldId = columnDescriptor.getPath().length == 1 ? 0 : primitiveType.getId().intValue();
+        int fieldId = path.length == 1 ? 0 : primitiveType.getId().intValue();
         int ordinal = schema.getType(columnDescriptor.getPath()[0]).getId().intValue();
 
         this.statsMap.put(
-            columnPath,
+            columnDotPath,
             new RowBufferStats(
-                columnPath, null /* collationDefinitionString */, ordinal, fieldId, primitiveType));
+                columnDotPath,
+                null /* collationDefinitionString */,
+                ordinal,
+                fieldId,
+                primitiveType,
+                clientBufferParameters.isEnableDistinctValuesCount(),
+                clientBufferParameters.isEnableValuesCount() && isInRepeatedGroup));
 
         if (onErrorOption == OpenChannelRequest.OnErrorOption.ABORT
             || onErrorOption == OpenChannelRequest.OnErrorOption.SKIP_BATCH) {
           this.tempStatsMap.put(
-              columnPath,
+              columnDotPath,
               new RowBufferStats(
-                  columnPath,
+                  columnDotPath,
                   null /* collationDefinitionString */,
                   ordinal,
                   fieldId,
-                  primitiveType));
+                  primitiveType,
+                  clientBufferParameters.isEnableDistinctValuesCount(),
+                  clientBufferParameters.isEnableValuesCount() && isInRepeatedGroup));
         }
       }
     }
+    statsFinder = new SubColumnFinder(schema);
```

Comment:
should only apply to iceberg land, so lets move this inside the previous brace.
Also you'll need a nullcheck in line 341.

Reply:
Made `SubColumnFinder` Iceberg mode only and added NPE check.